hoot-core-test/src/test/cpp/hoot/core/OsmMapTest.cpp

```diff
@@ -84,6 +84,11 @@ class OsmMapTest : public HootTestFixture
 
 public:
 
+  OsmMapTest()
+  {
+    setResetType(ResetBasic);
+  }
+
```

Comment:
The majority of the changes are adding a constructor that sets the reset type to `ResetBasic` for resetting map counters, UUID keys and the random number generator seed, or `ResetAll` that calls `TestUtils::resetEnvironment`.